src/btree/bt_debug.c

```diff
@@ -1015,17 +1015,19 @@ __debug_tree(WT_SESSION_IMPL *session, WT_REF *ref, const char *ofile, uint32_t
 static int
 __debug_page(WT_DBG *ds, WT_REF *ref, uint32_t flags)
 {
+    WT_BTREE *btree;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
 
     session = ds->session;
+    btree = S2BT(session);
     WT_RET(__wt_scr_alloc(session, 100, &ds->key));
 
     /*
      * Set up history store support, opening a history store cursor on demand. Ignore errors if that
      * doesn't work, we may be running in-memory.
      */
-    if (!WT_IS_HS(S2BT(session))) {
+    if (!WT_IS_HS(btree->dhandle)) {
```

Comment:
I believe this is basically the same as checking `session->dhandle`. If we do that, we don't need this `btree` variable since we already have a handle on the `session`.